frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,206 @@
+
+/* Component StorageClassDropdown - creates a dropdown list of storage classes */
+
+class StorageClassDropdown_ extends React.Component<StorageClassDropdownProps, StorageClassDropdownState> {
+    state = {
+      items: {},
+      name: this.props.name,
+      selectedKey: null,
+      title: <LoadingInline />,
+      defaultClass: null,
+    };
+
+    componentWillMount() {
+      this.componentWillReceiveProps(this.props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+      const { loaded, loadError, resources } = nextProps;
+
+      if (loadError) {
+        this.setState({
+          title: (
+            <div className="cos-error-title">Error Loading {nextProps.desc}</div>
+          ),
+        });
+        return;
+      }
+      if (!loaded) {
+        return;
+      }
+
+      const state = {
+        items: {},
+        title: {},
+        defaultClass: '',
+      };
+      const unorderedItems = {};
+      const noStorageClass = 'No storage class';
+      _.each(resources, ({ data }, kindLabel) => {
+        _.reduce(
+          data,
+          (acc, resource) => {
+            acc[resource.metadata.name] = {
+              kindLabel,
+              name: resource.metadata.name,
+              description: resource.metadata.annotations ? resource.metadata.annotations.description : '',
+              default: resource.metadata.annotations ? resource.metadata.annotations['storageclass.kubernetes.io/is-default-class'] === 'true' ? true : false : '',
+              accessMode: resource.metadata.annotations ? resource.metadata.annotations['storage.alpha.openshift.io/access-mode'] : '',
+              provisioner: resource.provisioner,
+              type: resource.parameters ? resource.parameters.type : '',
+              zone: resource.parameters ? resource.parameters.zone : '',
+            };
+            return acc;
+          },
+          unorderedItems
+        );
+      });
+
+      // Determine if there is a default storage class
+      _.each(unorderedItems, (item, name) => {
+        if ((item as any).default) {
+          state.defaultClass = name;
+        }
+      });
+      const { selectedKey } = this.state;
+      if (!state.defaultClass) {
+        // Add No Storage Class option if there is not a default storage class
+        unorderedItems[''] = { kindLabel: '', name: noStorageClass };
+      }
+
+      if (!this.props.loaded || !selectedKey ) {
+        state.title = <span className="text-muted">{nextProps.placeholder}</span>;
+      }
+
+      if (selectedKey) {
+        const item = unorderedItems[selectedKey];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      if (!selectedKey && state.defaultClass) {
+        const item = unorderedItems[state.defaultClass];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
```

Comment:
The logic is pretty much the same in these two `if` blocks. Can we make them one?

```js
const selectedItem = unorderedItems[selectedKey || state.defaultClass];
if (selectedItem) {
  state.title = this.getTitle(selectedItem.kindLabel, selectedItem.name);
}
```